Parse request method, URL, and protocol separately in web server logs.

aaronweeden opened the pull request:
<!--- The title text will be used to populate Changelog and associated documentation.
      Please make sure that the title is a complete sentence -->

## Description
<!--- Describe your changes in detail -->
<!--- Include screenshots for GUI changes if appropriate -->
<!--- If any documentation outside of this repo needs to be added or updated,
      please include links to the relevant docs and how they should be changed -->
This PR updates the web server log parser to split the request line (i.e., `%r`, e.g., `GET /index.html HTTP/1.1`) into its three parts (i.e., `%m %U %H`) so these can be parsed separately.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
This enables https://github.com/ubccr/xdmod-ondemand/pull/50.

## Tests performed
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
I added unit tests to make sure a log is correctly parsed whether the format is `%r` or `%m %U %H`. I also tested ingesting logs on my port on `xdmod-dev` with the changes from https://github.com/ubccr/xdmod-ondemand/pull/50 and the PRs on which it depends.

## Checklist:
<!--- Go over all the following points and make sure they have all been completed -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The pull request description is suitable for a Changelog entry
- [x] The milestone is set correctly on the pull request
- [x] The appropriate labels have been added to the pull request

jpwhite4 commented:
The webserver log parser already supports requestMethod (%m) and url === requestPath (%U) format strings. 

Why don't we use these?

jpwhite4 commented on classes/ETL/DataEndpoint/WebServerLogFile.php:
```diff
@@ -109,6 +109,10 @@ protected function decodeRecord($data)
         try {
             $decoded = $this->web_parser->parse($data);
 
+            $requestParts = explode(' ', $decoded->request);
```
In general can't guarantee that request is in the decoded output (e.g. could use "%m %U %H" instead of "%r") [in fact we probably want to do this in the ondemand module]

jpwhite4 replied:
If you really do want to do this I recommend a `property_exists` guard around it.

aaronweeden commented:
> The webserver log parser already supports requestMethod (%m) and url === requestPath (%U) format strings.
> 
> Why don't we use these?

@jpwhite4 Thanks, I have refactored it to use `%m`, `%U`, and `%H`; added unit tests; and updated https://github.com/ubccr/xdmod-ondemand/pull/50.
